src/stream/src/executor_v2/batch_query.rs

```diff
@@ -93,15 +96,16 @@ where
     /// Now we use hash as a workaround for supporting parallelized chain.
     fn filter_chunk(&self, data_chunk: DataChunk) -> Option<DataChunk> {
         let hash_values = data_chunk
-            .get_hash_values(self.info.pk_indices.as_ref(), CRC32FastBuilder)
+            .get_hash_values(self.key_indices.as_ref(), CRC32FastBuilder)
             .unwrap();
         let n = data_chunk.cardinality();
         let (columns, _visibility) = data_chunk.into_parts();
 
         let mut new_visibility = BitmapBuilder::with_capacity(n);
         for hv in &hash_values {
             new_visibility.append(
-                (hv.0 % self.parallel_info.degree as u64) == self.parallel_info.index as u64,
+                self.hash_mapping[(hv.0 % VIRTUAL_NODE_COUNT as u64) as usize]
+                    == self.parallel_unit_id,
```

Comment:
IIRC, hash_mapping stores vnode -> actor id, instead of parallel_unit_id?

Reply:
> hash_mapping stores vnode -> actor id, instead of parallel_unit_id?

Dispatcher's hash_mapping does, and it's mapped after `resolve_chain_node`. The original hash_mapping passed through context stores vnode->parallel_unit_id.